Mathlib/Data/Nat/GCD/Basic.lean

```diff
@@ -265,4 +265,12 @@ lemma div_lcm_eq_div_gcd (hkm : m ∣ k) (hkn : n ∣ k) : (k / m).lcm (k / n) =
   simpa [Nat.div_dvd_iff_dvd_mul (Nat.dvd_trans (Nat.gcd_dvd_left m n) hkm)
     (gcd_pos_of_pos_left n hm), Nat.gcd_mul_right m c n] using (Nat.dvd_gcd hmc hnc)
 
+lemma sub_eq_gcd_of_sub_dvd {p q : ℕ} (sub_dvd_q : p - q ∣ q) : (p - q) = p.gcd q := by
```

Comment:
I think the mathlib naming convention for hypothesis in argument is prefixed with `h` (not sure if this is enforced). In this case I think `(h : p - q ∣ q)` is fine

Also you can remove the extra parenthesis `(p - q) = p.gcd q ` -> `p - q = p.gcd q`

Perhaps also flip the equation direction to `p.gcd q = p - q`? Usually mathlib prefers the "complicated = simple" direction for equality lemma